newIDE/app/src/AssetStore/PrivateGameTemplates/PrivateGameTemplateInformationPage.js

```diff
@@ -266,56 +292,157 @@ const PrivateGameTemplateInformationPage = ({
+        return;
+      }
+
+      sendGameTemplateBuyClicked({
+        gameTemplateId: gameTemplate.id,
+        gameTemplateName: gameTemplate.name,
+        gameTemplateTag: gameTemplate.tag,
+        usageType: selectedUsageType,
+        currency: 'CREDITS',
+      });
+
+      const currentCreditsAmount = limits.credits.userBalance.amount;
+      const gameTemplatePriceForUsageType = privateGameTemplateListingData.creditPrices.find(
+        price => price.usageType === selectedUsageType
+      );
+      if (!gameTemplatePriceForUsageType) {
+        console.error(
+          'Unable to find the price for the selected usage type',
+          selectedUsageType
+        );
+        return;
+      }
+      const gameTemplateCreditsAmount = gameTemplatePriceForUsageType.amount;
+      if (currentCreditsAmount < gameTemplateCreditsAmount) {
+        openCreditsPackageDialog({
+          missingCredits: gameTemplateCreditsAmount - currentCreditsAmount,
+        });
+        return;
+      }
+
+      openCreditsUsageDialog({
+        title: <Trans>Purchase {gameTemplate.name}</Trans>,
+        message: (
+          <Trans>
+            You are about to use {gameTemplateCreditsAmount} credits to purchase
+            the game template {gameTemplate.name}. Continue?
+          </Trans>
+        ),
+        onConfirm: () =>
+          buyProductWithCredits(getAuthorizationHeader, {
+            productId: privateGameTemplateListingData.id,
+            usageType: selectedUsageType,
+            userId: profile.id,
+          }),
+        successMessage: <Trans>🎉 You can now use your template!</Trans>,
+      });
+    },
+    [
+      profile,
+      limits,
+      privateGameTemplateListingData,
+      gameTemplate,
+      onCreateWithGameTemplate,
+      isAlreadyReceived,
+      openCreditsPackageDialog,
+      selectedUsageType,
+      openCreditsUsageDialog,
+      getAuthorizationHeader,
+      onOpenLoginDialog,
+    ]
+  );
+
+  const getBuyButtons = i18n => {
     if (errorText) return null;
-    if (
-      isAlreadyReceived &&
-      privateGameTemplateListingData.includedListableProductIds
-    ) {
-      // Template is a bundle and is owned, no button to display.
-      return null;
-    }
 
-    const label = !gameTemplate ? (
-      <Trans>Loading...</Trans>
-    ) : isAlreadyReceived ? (
-      <Trans>Open template</Trans>
-    ) : isPurchaseDialogOpen ? (
-      <Trans>Processing...</Trans>
-    ) : (
-      <Trans>
-        Buy for{' '}
-        {formatProductPrice({
-          i18n,
-          productListingData: privateGameTemplateListingData,
-        })}
-      </Trans>
+    let creditPrice = privateGameTemplateListingData.creditPrices.find(
+      price => price.usageType === selectedUsageType
     );
-
-    const disabled = !gameTemplate || isPurchaseDialogOpen;
+    if (!creditPrice) {
+      // We're probably switching from one template to another, and the usage type is not available.
```

Comment:
I'm not sure I understand this comment?  Would the same component be reused for multiple templates?

Reply:
Yes, the state of "which item is selected" is managed at the top of the store, so when changing, this component stays mounted and keeps old state values. (I've tried updating it when listening to the template change, but maybe because setting state is asynchronous, I can't get it to work consistently. Which is why I've set up this catch (not ideal)